kcp/login.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+set -euo pipefail
+
+SCRIPT_DIR="$(
+  cd "$(dirname "$0")" >/dev/null
+  pwd
+)"
+
+KUBECONFIG_KCP="${KUBECONFIG_KCP:-$HOME/.kube/kcp.yaml}"
+KUBECONFIG_PLNSVC="${KUBECONFIG_PLNSVC:-$HOME/.kube/plnsvc.yaml}"
+WORK_DIR="${WORK_DIR:-$SCRIPT_DIR/work}"
+
+usage() {
+  echo "
+Usage:
+    ${0##*/} [options]
+
+Log onto the clusters running on KCP.
+
+Optional arguments:
+    -e, --environment ENVIRONMENT
+        Select an optional target environment (KCP and pipeline-service clusters).
+        The configuration templates must be located in a subdirectory of the
+        'environments' directory.
+        ENVIRONMENT must be in [$(ls -F "$SCRIPT_DIR/environments" | grep "/$" | tr '/\n' ', ')local]
+        Default value: local
+    -k, --kcp KCP_CONTEXT
+        Select the KCP instance on which to login. Must be one of [stable, unstable].
+        Default value: unstable
+    -w, --workspace KCP_WORKSPACE
+        Select the KCP workspace to log into.
+        Default value: default
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    ${0##*/} --all
+" >&2
+}
+
+parse_args() {
+  KCP_ENV="kcp-unstable"
+  KCP_WORKSPACE="pln-svc"
+  ENVIRONMENT="local"
+
+  local args
+  args="$(getopt -o dhe:k:w: -l "debug,help,environment,kcp,workspace" -n "$0" -- "$@")"
+  eval set -- "$args"
+  while true; do
+    case $1 in
+    -e | --environment)
+      shift
+      case $1 in
+      local)
+        ENVIRONMENT=$1
+        ;;
+      *)
+        ENVIRONMENT="$SCRIPT_DIR/environments/$1"
+        if [ ! -d "$ENVIRONMENT" ]; then
+          echo "Unsupported environment: $1" >&2
+          echo "" >&2
+          usage
+          exit 1
+        fi
+        ;;
+      esac
+      ;;
+    -k | --kcp)
+      shift
+      case $1 in
+      stable | unstable)
+        KCP_ENV="kcp-$1"
+        ;;
+      *)
+        echo "Unsupported kcp context: $1" >&2
+        echo "" >&2
+        usage
+        exit 1
+        ;;
+      esac
+      ;;
+    -w | --workspace)
+      shift
+      KCP_WORKSPACE="$1"
+      ;;
+    -d | --debug)
+      set -x
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    --)
+      # End of arguments
+      break
+      ;;
+    *)
+      echo "Unknown argument: $1"
+      usage
+      exit 1
+      ;;
+    esac
+    shift
+  done
+}
+
+argocd_local() {
+    mkdir -p "$WORK_DIR/argocd"
+    argocd --config "$WORK_DIR/argocd/config" "$@"
+}
+
+oc_kcp() {
+  KUBECONFIG="$KUBECONFIG_KCP" oc "$@"
+}
+
+oc_plnsvc() {
+  KUBECONFIG="$KUBECONFIG_PLNSVC" oc "$@"
+}
+
+copy_kubeconfig() {
```

Comment:
How does this function help if files (listed below) do not exist in the first place?

KUBECONFIG_PLNSVC:-$HOME/.kube/plnsvc.yaml
KUBECONFIG_KCP:-$HOME/.kube/kcp.yaml

Reply:
Good catch! A bit of code was missing when using an environment. However when using a local environment, we expect those files to be there as a prerequisite. The script that sets up the local env should make sure to create them.

Reply:
Could you please explain "this bit has been fixed"?
Do you mean that you have setup Gitlab and created these files there? If so, the README over there says to run this : 

```
git clone https://github.com/openshift-pipelines/pipelines-service.git
cd pipelines-service/kcp
git clone git@gitlab.cee.redhat.com:plnsvc/infra.git environments
./login.sh -e staging
```

Then, this public git repo can only be used to run local env? (both for ./login.sh and ./deploy.sh)

Reply:
The GitLab is used to create the kubeconfig files and populate them with authentication info.
Then deploy.sh can be used to setup those clusters.